src/vdom/diff.js

```diff
@@ -237,7 +237,9 @@ export function recollectNodeTree(node, unmountOnly) {
 
 /** Apply differences in attributes from a VNode to the given DOM Node. */
 function diffAttributes(dom, attrs) {
-	let old = dom[ATTR_KEY] || getRawNodeAttributes(dom);
+	ensureNodeData(dom);
```

Comment:
I think this will cause `setAttribute` to be carried on already-correct attributes.